[stayWefolio 서수연]  Add: footer 추가

suyeon-DV opened the pull request:
## 수정 사항 간략한 한줄 요약
[2020/10/20] Add: footer 추가
		 
## 수정 사항들 자세한 내용
[2020/10/20] Add: footer 추가
    - footer 레이아웃 추가
    - footer 스타일 추가

## 기타 질문 및 특이 사항
(없음)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

suyeon-DV committed:
Add: layout 작성
- 전반적인 layout 완성함

suyeon-DV committed:
Add: navbar layout 추가
- container layout 추가

suyeon-DV committed:
Remove: 주석 삭제
- 불필요한 주석 삭제

suyeon-DV committed:
Add: footer 추가
- footer 레이아웃 추가
- footer 스타일 추가

suyeon-DV committed:
Remove: 불필요한 파일 삭제
- dataSSY 폴더 내의 navdata.json 삭제

suyeon-DV committed:
Remove: 불필요한 폴더 삭제
- dataSSY폴더 내 data.json 파일 삭제

suyeon-DV committed:
Modify: JSX내 className 수정
- JSX내 class attribute를 className으로 수정

suyeon-DV committed:
Add: 미디어쿼리 추가
- 미디어 쿼리 추가
- 브랜드 로고 이미지 추가

suyeon-DV committed:
Modify: layout 수정
- menu layout 수정

claire-euni commented on src/components/Footer/Footer.js:
```diff
@@ -1,8 +1,52 @@
+            <span className="termInfo">이용약관</span>
+          </div>
+          <div className="officeInfos">
+            <span className="officeName">
+              - 상호 : (주)스테이폴리오 - 대표자 : 이상묵
+            </span>
```
* 반복적인 element들은 map()을 활용해서 코드를 간결하게 줄일 수 있습니다 ;) 
* className이 모두 다른데 다 다른 css 속성을 가지나요?

suyeon-DV replied:
반복적인 element 모두 map 활용해서 변경했습니다. 
두어개 정도 빼고는 다 같은 스타일입니다...ㅠㅠ 유의하겠습니다. 감사합니다.🥰

claire-euni commented on src/components/Footer/Footer.scss:
```diff
@@ -1 +1,90 @@
+
+$color-line: rgb(71, 71, 71);
+
+@mixin infoStyle {
+  color: rgb(153, 153, 153);
+  font-size: 14px;
+  line-height: 20px;
+}
+@mixin descriptionStyle {
+  padding: 2px 0;
+  color: rgb(71, 71, 71);
+  font-size: 12px;
+}
+
+@media (min-width: 768px) {
+  .container {
+    width: 750px;
+  }
+}
+@media (min-width: 992px) {
+  .container {
+    width: 970px;
+  }
+}
+@media (min-width: 1200px) {
+  .container {
+    width: 1170px;
+  }
+}
```
반응형 구현과 mixin 활용 베리굿🌟

suyeon-DV replied:
감사합니다.😊

claire-euni commented on src/components/Footer/Footer.scss:
```diff
@@ -1 +1,90 @@
+  .menus {
+    width: 100%;
+    @include flex-center;
+    li {
```
셀렉터 사이에 한줄 엔터 주세요~

suyeon-DV replied:
모든 셀렉터 사이에 엔터 넣었습니다!!

claire-euni commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,80 @@
+import "./Nav.scss";
 
-export class Nav extends Component {
+class Nav extends Component {
```
export default는 어디에 있나요........? Nav는 부를 수 없는 컴포넌트인가요?

suyeon-DV replied:
작동은 잘 되길래 어디갔나 했는데 지금은 밑에 'export default Nav;'가 있습니다. 
push 할 때 이상했었나봐요...ㅠㅠ 확인 잘 하겠습니다............

claire-euni commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,80 @@
+              </div>
+            </div>
+            <ul className="bottomLineWrap">
+              <li className="careby">
+                <span className="carebyLink">CARE BY</span>
+              </li>
```
이 부분도 map() 사용해보세요~

suyeon-DV replied:
모든 반복에 map 사용 유념하겠습니다. 모두 고쳤습니다☺

claire-euni commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,157 @@
+  right: 0;
+  z-index: 999;
+  @include flex-center;
+  .container {
```
셀렉터 사이에 엔터 한줄씩 주세요~

suyeon-DV replied:
모든 셀렉터 사이에 엔터 넣었습니다!!!! 감사합니다~!

claire-euni reviewed (commented):
코드가 반복되는 부분은 리팩토링 반영해서 간결하게 줄여주세요 화이팅! 🤗

suyeon-DV committed:
Modify: 저작권 문구 변경
- 각 info에 대한 문구 변경

suyeon-DV committed:
Refactor: 반복되는 부분 리팩토링
- 반복되는 element map 사용해 리팩토링

suyeon-DV committed:
Remove: 불필요한 주석 삭제
- 불필요한 주석 삭제

suyeon-DV committed:
Modify: 변수 사용 변경
- 색상에 변수 사용하도록 변경
- 가상선택자 사용해 레이아웃 변경

suyeon-DV committed:
Refactor: 반복되는 li 리팩토링
- 반복되는 li를 map 사용해서 리팩토링

saengmotmi commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,157 @@
+      }
+    }
+    .navContainer {
+      width: 100%;
+      height: 100%;
+      display: flex;
+      flex-direction: column;
```
다른 PR에 남겨드린 내용 확인하시고 고칠 수 있는지 검토 부탁드려요~

suyeon-DV replied:
네!!! 앞선 리뷰보고 고칠 수 있는 것 고쳤습니다. 
- 불필요한 width, height:100% 모두 삭제
- mixin에 argument 이용해서 적용
- classname의 동적사용
이렇게 세가지 수정했습니다. 
하지만 mixin 사용하면서 common.scss에 있는 mixin에 flex-direction을 마지막에 넣어서 다른 flex-direction: column도 수정하려고 했는데 앞선 인자를 건드리지 않고 3번째 인자만 사용하는 방법을 찾아보았는데 계속 complie에러가 나서 사용하지 못 했습니다. 나중에 찾아뵙겠습니다ㅠㅠ
<사용하려했던 코드>
```
@mixin flex($justify-content: center, $align-item: center, $flex-direction: row) {
  display: flex;
  justify-content: $justify-content;
  align-items: $align-item;
  flex-direction: $flex-direction;
} 

// 시도해본 코드
// 1번
@includes flex(null, null, column)

// 2번
@includes flex(3th (column))
```

2번은 '둘 이상의 인수가 있고 두 번째 인수에 액세스하려는 경우에는 propertyname : nth(variable_name, 2) 를 작성하여 수행 할 수 있습니다.'
라는 말을 보고 시도해봤는데 제가 못 이해한 것같기도 합니다....🙄😭

saengmotmi commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,157 @@
+            color: $color-title-black;
+            font-size: 10px;
+          }
+          .loginLink {
+          }
+          .loginAnd {
+          }
+          .signupLink {
+          }
+        }
```
필요한 내용일까요?

suyeon-DV replied:
삭제했습니다!!

saengmotmi commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,157 @@
+        padding: 0 2px;
+        border: 1px solid $border-gray;
+        display: flex;
+        justify-content: space-around;
+        li {
```
새로운 셀렉터가 시작할 때는 매번 한줄씩 띄워주세요

suyeon-DV replied:
셀렉터에 한 줄씩 띄웠습니다.

saengmotmi commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,80 @@
+                      <i class="fab fa-instagram"></i>
+                    </a>
+                  </li>
+                  <li className="naver"></li>
```
셀프클로징 태그 적용해주세요

suyeon-DV replied:
수정했습니다!

saengmotmi commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,80 @@
+            <div className="topLineWrap">
+              <form className="searchBarWrap">
+                <input className="searchBar" type="text" placeholder="Search" />
+                <button className="searchBarBtn">
```
```suggestion
                <button type="button" className="searchBarBtn">
```
button 태그에 type button을 부여하는 이유를 찾아서 댓글로 남겨주세요!

suyeon-DV replied:
찾아보고 블로그도 썼습니다!!! 
button은 기본값으로 type="submit"을 가지고 있습니다. form 태그 내부의 button 역시 기본값은 submit인데 이 경우 버튼을 눌렀을 때 의도치 않은 동작 즉, 버튼은 submit을 수행하게 됩니다. 
하지만 type="button"일 경우 버튼은 특정 이벤트를 걸어주지 않는한 아무동작도 하지 않습니다. 그렇기에 버튼의 의도치 않은 동작을 예방할 수 있어서 꼭 type="button"을 써야합니다. 

또한 form 태그 밖의 button은 기본값으로 submit을 가지고 있지는 않지만 sementic tag로서 form 태그가 권장되기 때문에 반드시 type="button"을 쓰는 습관을 들여야합니다. 

제가 기본기가 부족했음을 느끼고 있습니다. 리뷰 감사합니다😍

saengmotmi commented on src/components/Footer/Footer.scss:
```diff
@@ -1 +1,90 @@
+
+  .footerInfos {
+    width: 1170px;
+    margin-top: 30px;
+    margin-left: auto;
+    margin-right: auto;
+    padding-top: 20px;
+    padding-left: 15px;
+    padding-right: 15px;
```
이정도라면 단축 속성으로 써주는게 좋지 않을까 싶네요!

suyeon-DV replied:
단축 속성으로 이용했습니다!!

saengmotmi commented on src/components/Footer/Footer.scss:
```diff
@@ -1 +1,90 @@
+}
+@mixin descriptionStyle {
+  padding: 2px 0;
+  color: rgb(71, 71, 71);
```
위에 선언해둔 변수가 있는 것 같은데 사용이 안되는 것 같습니다

suyeon-DV replied:
헉... 전 왜 안 보이는 걸까요.... 변수 사용했습니다!!! ☺

saengmotmi reviewed (commented):
리뷰 확인해주세요 :)

suyeon-DV committed:
Refactor: 멘토 리뷰 반영
- mock data 따로 분리
- mixin을 변수 이용에서 사용하도록 변경
- 스타일 적용되지 않는 nesting 삭제
- width, height 100% 모두 삭제 후 각 요소가 공간을 차지하도록 수정
- 공통된 스타일이 적용되는 className 삭제 후 하나의 Tag만을 사용하도록 수정
- className의 동적사용에 따라 스타일이 변경되도록 수정

suyeon-DV committed:
Modify: mixin 내용 수정
- mixin에 인자 사용하도록 수정

suyeon-DV committed:
Add: feature/navlayout 병합
- feature/navlayout 병합

suyeon-DV committed:
Refactor: 멘토 리뷰 반영
- 불필요한 height:100% 삭제
- arguments 이용한 mixin 사용
- classname의 동적 사용을 통한 스타일 변경

suyeon-DV committed:
Modify: 셀렉터 사이 엔터 추가
- 셀렉터 사이 엔터 추가

suyeon-DV committed:
Remove: 불필요한 파일 삭제
- nav.js 및 nav.scss 파일 내용 삭제

suyeon-DV commented:
뒤늦게.... 예전에 작업했던 feature/navlayout 브랜치의 내용이 해당 브랜치에 포함되어 있는 것을 발견하고 모두 삭제했습니다ㅜㅜ
종택님 리뷰 보시느라 정말 고생많으십니다😭😭 좋은 공부가 되고 있습니다!!! 🤗🤗

suyeon-DV committed:
Modify: mixin 수정
- 모든 flex 속성 mixin 사용해 수정

saengmotmi commented:
LGTM 🔥
